File: tai-orm/src/main/kotlin/tai/orm/core/BuilderContext.java
package tai.orm.core;

/**
 * Created by sohan on 3/28/2017.
 */
public interface BuilderContext<T> {

    BuilderContext<T> putEmpty(String key);

    boolean isEmpty(String key);

    boolean contains(String key);

    T get(String key);

    BuilderContext<T> put(String entity, T upsertEventDispatcher);

    BuilderContext<T> makeImmutable();
}

File: tai-orm/src/main/kotlin/tai/orm/core/BuilderContext.kt
package tai.orm.core

/**
 * Created by sohan on 3/28/2017.
 */
interface BuilderContext<T> {
    fun putEmpty(key: String): BuilderContext<T>
    fun isEmpty(key: String): Boolean
    operator fun contains(key: String): Boolean
    operator fun get(key: String): T
    fun put(entity: String, upsertEventDispatcher: T): BuilderContext<T>
    fun makeImmutable(): BuilderContext<T>
}